### ISSUE TITLE
Foundation 6: component feedback

### ISSUE THREAD

**Issue by gakimball** [zurb/foundation#6194]
Foundation has about two dozen components and half as many JavaScript plugins. This number has grown steadily since the framework was first introduced, but for version 6, we're interested in scaling back slightly. Some components may be combined and streamlined, while others may be deprecated entirely or maintained as a separate codebase.

Here's an initial list of components that could be deprecated:

- Block grid (merged into grid)
- Top bar (replaced with something better)
- Orbit
- Clearing
- Pricing table

Conversely, here are a few of the most common requests for new components:

- Date/time picker
- Mega menu
- Autocomplete/typeahead
- Flexbox grid
- Sticky nav (like Affix or Sticky-kit)

**We'd like to hear anything and everything you have to say about components here.**
 - Which ones do you use often?
 - Which ones do you never use?
 - What features would you like to see in existing components?

If you have feedback on *how* the components are written, from an HTML/Sass/JS perspective, we have feedback topics for that as well.

**[View our other discussion topics for Foundation 6 here.](https://github.com/zurb/foundation/labels/Foundation%206)**

**Comment by datenimperator** [comment]
I've used the pricing table twice, it came in handy. Deprecating it wouldn't hurt too bad, but still require me to work a little more.

**Comment by zurbrandon** [comment]
@datenimperator For things that specific, we're thinking of keeping them outside the main library, but having a place where you can grab the Scss and html to make them easily. We have used them plenty, but know that for at least 50% + of our sites we don't.

Thoughts on that?

**Comment by datenimperator** [comment]
@zurbrandon Keeping them outside the main lib would be fine, but that'd be true for almost all of the components. Currently I try to manually disable css and js that I don't use, would be nice if there'd be some way to do this more elegantly.

**Comment by zurbrandon** [comment]
In the Sass it's pretty easy, but JS gets a little harder. That will be a large push for this build.

**Comment by gakimball** [comment]
@datenimperator With Foundation for Apps we did rework this slightly. Every component is contained in one map called `$include-css`; any value listed as `true` will get CSS classes exported.

```scss
$include-css: (
  grid: true,
  button: true,
  tabs: false,
);
```

This allows you to disable CSS output, but still have access to the mixins, and not have to mess with a giant list of `@import` statements. You can just import the entire framework.

**Comment by bcartier** [comment]
I'd like to see a slimmed down list of components, but have each component evaluated by a third party accessibility expert (or organization). Test the components with actual users, then make the components as accessible as possible out of the box, and expand the component docs with detail on how to implement components together in an accessible way. 

I'd like Foundation 6 to be the definitive answer to "what's the best framework for building accessible websites?". 

Tall order, I know ;-)

**Comment by zurbrandon** [comment]
@bcartier Like that thinking! That will for sure become a priority on this build. Are you pretty knowledgeable on accessibility?

**Comment by izakfilmalter** [comment]
Maybe we can create a component API that allows for custom components to be built into Foundation 6 that are shipped as individual packages, ala bower dependencies. You could then pick and chose which ones you wanted and visit maybe a public index on foundations site of different components and grab what you need and then move on. This would allow you to focus on the 5 or so great components that you make, split them off into their own repos with separate issues. Also this could allow you to fix a bug in a component without having to push a release of Foundation. Other people have requests for a specific component and they would then be easy to build. This would work especially well with moving the settings into a map.

**Comment by ksherman** [comment]
Concerned about deprecating the block grid, I rely on it a ton for repeated items in <li>'s, such as a grid of headshots that have different layouts at different screen sizes. The regular grid doesn't really do this. Maybe I would like to see the block grid have equal heights in each row? I think equalizer was expanded for this, if so, ignore that :)

Definitely on board with a mega menu option! 

Honestly I use fewer and fewer of the components as time has gone on. How about some form of responsive typography option?

Flexbox is super interesting and I can't wait for it! But I'm one of those lame people that have to try and support older IE versions, which is tough (even more annoying when people use newer versions of IE in compatibility mode, I've stopped trying to accommodate them). So I would certainly appreciate a parallel version with the float grid for a while.

Here's my list, though note that I just work on one project full time, and just never had a need for some of these or I have my own versions based on our needs.
##### Never use:
* Pagination
* Side Nav
* Icon Bar
* Breadcrumbs
* Sub Nav
* Dropdown Buttons
* Split Buttons
* Switches
* Keystrokes
* Inline Lists
* Tables
* Pricing Tables
* Thumbs
* Accordion
* Alert Boxes
* Clearing
* Dropdown
* Interchange
* Joyride
* Magellan
* Off Canvas
* Reveal
* Slider
* Tabs
* Tooltips
* Top Bar

##### Use all the time:
* Grid
* Block Grid
* Buttons
* Button Groups
* Forms
* Type
* Labels
* Visibility
* Flex Video
* Abide
* Equalizer
* Orbit (super sad about this one, I've still got some work to do to get Slick working)

**Comment by tipsforthings** [comment]
I'd say refresh the navigation system. As you know, the sidebar I've designed looks pretty cool. I think if this was implemented into Foundation, that would be pretty amazing.

As for setting the whole thing up, you could have 2 classes for the sidebar, 'sidebar' and 'sidebar-icons'. On top of this you could assign sidebar id's of 'sidebar-left' and 'sidebar-right', to have dual sidebars that collapse on small displays.

For the standard sidebar, I'd say that keeping the top bar would be the way to go, but when the sidebars are setup with the clases of 'sidebar-icon' then have the top bar disabled and use arrows to expand the navigation, as you can see on my site. http://alxs.co.uk

**Comment by zurbrandon** [comment]
Hey @ksherman We're not thinking of removing the block grid function, but removing it as it's own component. Currently with Foundation for apps we've combined the idea with the regular grid and called it parent sizing. 

So instead of: 

```
<div class="row>
  <div class="small-12 columns>
    <ul class="small-block-grid-3">
      <li> Content</li>
      <li> Content</li>
      <li> Content</li>
    </ul>
  </div>
</div>
```

You could have:

```
<div class="row small-3-up">
  <div class="column"> content</div>
  <div class="column"> content</div>
  <div class="column"> content</div>
</div>
```

We feel like this gives the block grid functionality more context and allows us to deal with gutters more consistently.

**Comment by ksherman** [comment]
Hmm, okay. Would this also work and show a grid, six across on medium three across on mobile? I honestly haven't spent time (yet!) with Apps to know the syntax.

```
<div class="row small-3-up medium-6-up">
  <div class="column"> content</div>
  <div class="column"> content</div>
  <div class="column"> content</div>
  <div class="column"> content</div>
  <div class="column"> content</div>
  <div class="column"> content</div>
  <div class="column"> content</div>
  <div class="column"> content</div>
  <div class="column"> content</div>
</div>
```
I definitely have gutter issues mixing the grids together from time to time, so I totally get the motivation.

**Comment by zurbrandon** [comment]
@ksherman Yeah, that would totally work with it.

**Comment by Anhellius** [comment]
Definitly a mega menu, that´s actaully with what i´ll be working for the next couple of days.
I know it wasn´t listed, but i would like to know what you think about implementing material design, as a complement maybe ?
thanks

**Comment by Lynda333** [comment]
Here are some of my thoughts:

On the list you provided:

    Block Grid being merged into the grid sounds great as long as I can still easily make block grids.
    Top bar replaced with something better sounds fine to me. I do use the TopBar in virtually all of my projects. There have only been a couple that I haven;t used it in and those were one page sites.
    I never used Orbit as I used more powerful sliders in my projects. (LayerSlider is the one I like and recommend but it is a paid file.) Clearing I have never used.
    I have used Pricing Table one time and it was useful but isn't something I would use often. 

    Date/time picker - Don't use them that often but I could see the benefit.
    Mega menu - Would love to see this. I managed to build one following info on the Foundation Forums but one that is built in would be awesome!
    Autocomplete/typeahead - Possibly but unsure.
    Flexbox grid - Would love to see flexbox in F6!
    Sticky nav (like Affix or Sticky-kit) - YES PLEASE! I really could use this. 

Here are some thoughts on each item:

    Grid - I would really like to see flexbox in F6. I realize there may be the need for a floated grid for support of some browsers but having it available for those of us who want to use it I really hope happens. =>
    Interchange - I never have used it but I can see the usefulness. 

    Off Canvas - I would love to see an option to have Off Canvas merged into a menu system or at least a way to have an off canvas menu when on small devices as part of the main menu. I know some people have created a way to do this but having it baked in and easy to setup would be very nice. 
    Top Bar - I use the Top bar in almost every project. Having a redesigned top bar that is even better I would love to see. If it includes a mega menu and other options then it gets even better! Also a way to quickly center the top bar navigation (or whatever it would be called) needs to be included. I have manually done this but having it included and easily doable is the goal. I have an open request (#4885) on this topic.
    Icon bar - I have used this in a project recently and I like it. I will probably start to use it more.
    Side nav - I use it on many sites but having improvements on the side nav would be awesome. I know @tipsforthings designed a sidebar and I like it a lot. My only concern is the icons on the small view. While I use and love icons not all icons are universal and icons only could lead to some confusion for my end users so text on small would still be needed.
    Magellan Sticky Nav - I have used it a couple times but would rather see it merged into a sticky feature that will allow me to stick any item on a page. I could then make a sidebar that is just like magellan sticky nav but it is just a sidebar with the sticky class applied to it.
    Sub nav - haven't used much.
    Breadcrumbs - I use breadcrumbs on several sites and will continue to use them. Very easy to integrate with foundation.
    Pagination - Don't use it much because my CMS handles this for me already so I don't use Foundation's stuff. 

    Orbit - I always use other sliders.
    Thumbnails - I really don't use this one either.
    Clearing - Never use it.
    Flex video - Use it all of the time! LOVE this one! 

    Forms - I use forms on every site and having styling baked in is great.
    Switches - Never used them so far.
    Range Sliders - I have used Range Sliders but not Foundations. I only used them with one project. The range sliders I use are:  noUiSlider  - http://refreshless.com/nouislider/ - I was using these before Foundation added them to the framework. noUiSlider also has lots of options and is easy to work with.
    Abide - Use it all of the time. Any improvements with Abide would be great but I don't have anything specific to mention! 

    Buttons - Use them all of the time. Would like to see additional styling options such as outlined buttons, buttons with shadows, etc. that I can quickly integrate into the site.
    Button Groups - Very useful to have. Again more styling options.
    Split Buttons - Never really used them.
    Dropdown Buttons - Use them sometimes and they are useful.
    A note on buttons: Other than the styling mentioned above I have started using a premium script called [Morphy Buttons] (http://codecanyon.net/item/morphy-buttons-jquery-any-content-morpher/8253253?ref=ubhape2) that takes a button and will morph that button, when clicked, into a modal or lightbox. It is a cool effect and a great way to show content. Something like that would be cool to see in Foundation. 

    Type - Great to see some baked in styles for type. I would like to see some responsive typography options included. I have a few instances where I really need this.
    Inline List - Use it all of the time. Very useful.
    Labels - Don't use it often but it is useful.
    Keystrokes - Again don't use much but every now and then.

    Reveal Modal - Use all of the time.
    Alerts - Use occasionally. Useful when I do use them.
    Panels - Don't use them that often.
    Tooltips - Use them often. Would love to see a way to easily add an image into the tooltip.
    Joyride - Love it! I have started to use this more and more.

    Dropdowns - Useful feature. I use it here and there.
    Pricing Tables - Rarely used them but are useful when needed.
    Progress Bars - I use them with one project and would use them more if I had a place to use them. I like the simplicity of the progress bars.
    Tables - Use them occasionally when needed. Good to have. 
    Accordion - I use it quiet often. It is easy to use and very useful.
    Tabs - Same as accordion. Use it quite often.
    Equalizer - This has been a God send! Really, really nice to have. 

I really like the idea of keeping some of the components outside of the main framework but something you can quickly grab if needed. I also like the idea of components that can be used as a stand alone item vs having the framework on the site.

Thank You!

**Comment by bcartier** [comment]
@zurbrandon Unfortunately, I'm far from an expert. I am finding that clients are becoming more aware of accessibility issues, and now expect sites to be accessible (rightly so). For many markets/organizations it's mandatory. 

I might suggest collaborating with community over at [The Accessibility Project](http://a11yproject.com/)? Maybe if Zurb were to announce a commitment to making v6 the most accessible framework, you might get some contribution directly from their community?

**Comment by joelkinzel** [comment]
+1 for accessibility being a huge push. I work in the public sector and this is a requirement for us. 

Three big wins IMO would be:

1. Mega menus. Certain other frameworks suggest them but then don't really provide a means to building them.
2. Off-canvas being multi-tiered. Right now off-canvas is a single tier, or perhaps a sort of nested looking list. It would be awesome if it could slide to another level for larger more complex websites. 
3. A media object component - it is extremely useful in many, many cases

If you do drop Orbit, maybe include a third-part carousel would be nice, or at least an example of one that you recommend somewhere in the docs.

**Comment by JeremyEnglert** [comment]
Glad to see that certain items will be taken out of the core - things such as the pricing tables are so specific that they are only useful to a small subset of users. It would be cool to see a page that contains code for things like pricing tables, clearing and slideshows. This would also encourage the Foundation community to share the components they create for their websites with everyone.

As far as the top bar goes, I am EXTREMELY happy to see that is going to be reworked. It definitely has it's benefits, such as has scaling between desktop and mobile, however, it is extremely rigid and hard to work with. Every website is going to have a navigation of some sort, however, the top-bar is best suited as just that - a top bar. If you want to use it on another part of the page or include a logo, it is very difficult to work with. I'd say the large majority of my time is spent on overriding the styles/functions of the top bar.

As stated above, every website is going to have a navigation. Therefore, it would be nice to see multiple types of navigation options offered. For example, the "multi-level" mobile nav is cool, but if you ever want to have something more conventional, it is pretty difficult to change and involves hacking up the JS files. 

tldr; Topbar is hard to work with. Need more navigation options that are easy to work with.

**Comment by gakimball** [comment]
@Anhellius We're looking into incorporating the motion library we wrote for Foundation for Apps into this framework, but we probably won't take any elements from Material wholesale. The framework needs to stay light and unopinionated.

@Lynda333 Thanks for the super thorough feedback! :)

@bcartier Thank you for the link, that's a really good resource.

A few of you mentioned top bar, and better responsive navigation, and the tricky thing there is balancing magic with complexity. Top bar is great because it converts to mobile with no configuration, but you have to follow the rules of the HTML precisely, and good luck changing anything with how it works. But it's something we've been talking about.

In terms of navigation, there's probably an opportunity to consolidate components. For example, in Foundation for Apps we built a [flexible menu bar](foundation.zurb.com/apps/docs/#!/menu-bar) which supports icons or text (or both), can be oriented horizontally and vertically, and can be stretched or compacted. The one big thing it doesn't support right now is dropdowns. But it's a good example of a component that can accommodate multiple use cases: top bar menu, off-canvas menu, icon bar, simple inline list, and more.

**Comment by christoffervivian** [comment]
You can't add a border to Top bar without it breaking, which sucks. 

Foundation for Sites 6 should be simpler and get rid of components that aren't widely used, at least move some of them like Joyride, Switches, Range Sliders, Clearing, Orbit, Pricing Tables and Icon Bar out of the core framework. Foundation used to be so simple and non-bloated, I remember when you guys wouldn't even use SCSS. Please go back to your roots. Bootstrap is looking way easier to use now a days.

**Comment by JeremyEnglert** [comment]
An all CSS/Sass navigation would be an interesting route to take as well. As I've mentioned before, you currently have to modify the JS to make major changes to the topbar.

**Comment by tipsforthings** [comment]
On thing I've noticed is joyride isn't very 'mobile-friendly'.  It would be great if joyride was able to be configured seperately for mobile devices.

@Lynda333 you know the text is still there on the side bar, it is easily adaptable to hide the icons but if you press the arrow at the top, the full sidebar slides out, off-canvas style.

**Comment by gakimball** [comment]
@tipsforthings In what ways do you think it could be improved for mobile?

**Comment by proimage** [comment]
My 2 cents:

- Media Queries: Keep.
- Visibility: I use but mainly for making form labels responsive... it feels like a hack every time. Some type of `<label class="large-up-right large-up-inline">` thing would resolve things for me, allowing labels-above-fields for `small` and `medium`, and labels-next-to-fields for `large` and up.
- Grid: KEEP.
- Block grid: Keep functionality. Merging into Grid is ok if the classes can be named sensibly. Not sold on `.small-3-up`.
- Interchange: Keep, but needs ability to combine named media queries (eg. `data-interchange="[image_path, (small retina)], [image_path, (small)]"`).
- Utility: Love 'em. Keep.
- Javascript Utils: Never used 'em, but I'm not a JS guy.
- RTL Support: Haven't used it yet, but very glad it's there... I'm based in Israel. :)
- Off-canvas: Keep. Could use more flexibility of the off-canvas content (less imposed styles). Allow transformer-style conversion to a mega-menu for `large-up` and you have a winner.
- Top bar: Very useful, but doesn't seem as flexible as the rest of F5. Doesn't like `base-font-size` changes in my experience.
- Icon Bar: Keep.
- Side Nav: Meh... doesn't really do enough to justify being a thing IMO.
- Magellan Stick Nav: Keep. Perhaps enhance with a way to pad the bottom of a page with white-space so it can be scrolled down far enough to place those short nav anchors we have at the end of pages at the expected position at the top of the window?
- Sub nav: Doesn't do all _that_ much as-is. Could be more useful if integrated with dropdown/split buttons IMO.
- Breadcrumbs: Keep. Could be nice to allow for the last crumb, which sometimes represents the current page, to be an `<h1>` element.
- Pagination: Keep. Would be nice to have `|<< First` and `Last >>|` options.
- Orbit: I've converted to Slick already. Understandable deprecation, if a bit sad. \*sniffle\*
- Thumbnails: Nice, but no real need for it.
- Clearing Lightbox: Keep, but needs enhancements - it's kinda jarring as-in. Fade it in, space the thumbnails out, etc. 
- Flex Video: Keep, but rename to something like "Flexible Frames" or whatever; it's not CSS3's `flex`, and it's useful for other embedded content besides just videos - like Google Maps. Make width/height ratio selection more dynamic (hooking into `data-ratio="16:9"` would be fine by me).
- Forms: Where's the love for multi-select fields??!? I know - I hate them too, but sometimes the CMS insists on outputting them. Perhaps F6 could convert them to a height-limited scrolling list of checkboxes (which is easier to understand anyway)?
- Switches: Keep. Never used them, but I should start. Maybe allow adding "On" / "Off" type text to the sides?
- Range Sliders: Keep, but hook into the existing HTML5 `<input type='range'>` element: http://demosthenes.info/blog/757/Playing-With-The-HTML5-range-Slider-Input
- Abide: Keep - I love it!
- Buttons: Keep.
- Button Groups: Keep.
- Split Buttons: Keep, but perhaps keep the dropdown toggling part highlighted as long as the dropdown is open?
- Dropdown Buttons: Same remark as split buttons.
- Button Groups: Keep.
- Type: `<small>` inside headers is a wee bit too small IMO.
- Inline Lists: Keep.
- Labels: Keep. Give those sweet Foundation Icon Fonts some styling for when they're placed  inside these.
- Keystrokes: Never used them. Visually nice, but kinda pointless IMO since all it does is style the `<kbd>` element. Besides, aren't keystrokes more of an App thing?
- Reveal: Use all the time. Would be nice to have native `<header>` and `<footer>` styling for them...
- Alerts: Keep... use 'em all the time.
- Panels: Meh.
- Tooltips: Needs better auto-positioning (or did this already happen?). For manual positioning, keep in mind that there are 2 "positions" to take into consideration - the point on the triggering element that the tooltip will be attached to, and the corner or side of the tooltip to attach to the aforementioned point.
- Joyride: Keep.
- Dropdowns: Keep. They could stand to have the same positioning logic (both automatic and manual) applied to them as tooltips.
- Pricing Table: Demote to a first-party plugin. ;)
- Progress Bars: Keep, but like with the auto-complete, use the native HTML `<progress>` element as a basis to build upon (http://css-tricks.com/html5-progress-element/). Things I've needed are: text attached to the start and end of the entire meter, and to the current position along the meter; text attached to specified points along the meter (think "stretch goals", "halfway markers", etc), styled with tooltip-like arrows marking the exact spot; and animations (not sure if this should be left to us to develop, but they seem like sensible baseline styling for an element that indicates on-going activity... you wouldn't give us a spinner component without the actual rotating effect, would you?)
- Tables: Keep. Put the even/odd striping into its own class name, like `<table class="stripped">`, since sometimes having it striped just doesn't make sense. Aside from that, it would be great to integrate some sort of flexible/responsive table solution (I've had to revert to faking my responsive tables using the grid system - talk about turning the... err, *tables* on table-based layout!).
- Accordion / Tabs: Keep, but where did the auto-switching from one to the other go? Tabs used to auto-collapse into an accordion in F4 (?)... Didn't I see you guys retweeting someone's 'tabcordion' a couple of weeks ago? :)
- Equalizer: KEEP. ;)

Suggestions for new components:
- Mega-menu: heck yes. Possibly convert it to off-canvas for small screens (or, since we're doing mobile-first these days, allow the off-canvas menus to auto-convert into mega-menus for larger screens)... kinda like I did here, but betterer: http://codepen.io/sandwich/pen/awpiq
- Auto-complete/type-ahead: Yes, but definitely build off the existing (but kinda ugly) HTML5 `datalist`: http://viralpatel.net/blogs/html5-datalist-example/
- Button Rows: Needs a better name, but think about those table rows where the entire thing was a clickable link. Now recreate them with the grid system instead of tables for responsiveness, and perhaps add a nice right-pointing chevron/arrow on the right end, and you've got it. I did up a quick'n'dirty demo here: http://codepen.io/sandwich/pen/ogWqaY

**Comment by gosupernova** [comment]
Personally I am seeing myself using less components in foundation and trying to have a lighter touch when it comes to responsive sites. So I will list the ones I think I use the most:

- The grid
- Tabs (would be nice if they went from tabs to accordion again)
- accordian
- Offcanvas
- Interchange
- Modals
- Equaliser
- Slick slider (hat tip this is awesome!)

New ones I would use
- Mega menus (sometimes)
- Sticky sidebar (spent a lot of time that one)
- flexbox grid 
- autocomplete
- date picker  

I would love to see a section on your site where there are a bunch of third party ones which have passed certain standards (foundation approved) and give a solid range of responsive pattern solutions that would rock - a components library, this might enable an even more scaled back list and support would be done by the authors.

**Comment by veego** [comment]
I think you can include your awesome font icons to next foundation, I use that on every project (I'm sure that I'm not alone).

**Comment by proimage** [comment]
I'm split about the icon fonts actually. SVG is actually the way forward with regards to icons, but it typically takes more effort to implement. I end up using Foundation's icon fonts when I'm being lazy or am rushed for time, but when I can allow myself to do things properly, I use SVG icons (custom picked from IcoMoon, usually).

I'm fully behind graduating the icon fonts out of beta, but perhaps to a first-party add-on instead of integrated into the core.

**Comment by notoriousb1t** [comment]
I love the icons, but I would like to petition against including icons in the default full build.  Some companies have policies against forking or doing custom builds for 3rd party libraries.  If you don't need icons for the project or if you need to use a custom icon set, this is just another extra thing to carry around.  One of the things that turns me off about Bootstrap is that they include glyphicons in the default build.

**Comment by adamj88** [comment]
What about merging sub-nav and side-nav into a singular component for vertical/horizontal nav? If you plan to rewrite the topbar then perhaps they could all go into a navigation component?

The inline list component isn't too dissimilar to a subnav either, they're both inline lists of links effectively.

I was sad to see the depreciation of orbit, it was one of my most used components.

One thing I'd like to see is offcanvas have more options, e.g. top & bottom or multiple of each like foundation for apps, and the offcanvas tab bar have support for multiple icons in its sections.

**Comment by gakimball** [comment]
@adamj88 With the navigation stuff, we're leaning in that direction. If you look at the [menu bar](http://foundation.zurb.com/apps/docs/#!/menu-bar) we wrote for Foundation for Apps, you can get an idea of what a more universal nav component might look like. We just have to make it work without flexbox ;)

The off-canvas component will most likely become just an empty box that you place other things inside. This will make it *way* more flexible than the current one, which is basically only for menus.

Orbit may stick around in a more simplified form, but that's still up in the air. For certain kinds of components which are a dime a dozen, like carousels, lightboxes, and so on, we're wondering if it's worth competing with established plugins that have tons of features.

**Comment by andrewjsaint** [comment]
+1 for the Susy-esque map settings.
+1 for the "empty box" off-canvas.

What should go:

- Data attributes like `<form data-abide>` (doesn't play nice with Wordpress) to call JS.
- Orbit
- Clearing Lightbox
- Icon Bar, feels more suited to F4Apps.
- LTR (something like css flip instead)

What I think should go:

- Equalizer (replace with Flexbox?)
- Switches
- Thumbnails

What you should add:

- Animations / Transitions.
- 3rd party extensions repository.
 
What I think should change:

- A more uniform and simplified approach to spacing, similar to Bootstrap that basically uses @line-height-computed for margins on all components.
- A more uniform and simplified approach to settings in general. For example declaring borders, it doesn't need 3 variables just the 1 in shorthand. Also variables like $secondary-color feel underused.

**Comment by gakimball** [comment]
@andrewjsaint To your point about using Sass maps to configure components, we'd like to, but there's one big caveat which is that you can't reference previous values in a map while it's being defined. [I outlined the situation here](https://github.com/zurb/foundation/issues/6196#issuecomment-70303599); do you have any ideas for how we could get around it?

**Comment by rafibomb** [comment]
Just want to be clear, there may be components moved out of the core, but will be easily accessible to add in as needed. The end result will actually be more available components while the core framework contains the essentials.

**Comment by erredeco** [comment]
I'll try so spare my 2 cents, just some considerations ;)

1. In general, I think it could be useful to separate the framework and the components from the "Theme" (with "theme" mean where the appereance, that is for example the colors, etc.)

2. The typography: cardinal.css uses a responsive font-size that depends on the viewport. Opinions?
In general I think that the font size of every element should be inherited from the body element and it should be explicitly declared only in the case you overwrite the value with a different one. (same for line-height)

3.  about the buttons, we have 5 styles and 4 classes: (default), .success, .secondary, .info, .alert
...honestly, I never used all of them altogether. Could it be a good idea to build all of them via the general button mixin? BTW:  the button mixin should let me choose the text color ;)

4 +1 for the "empty box" off-canvas; that would be nice to have both  the off-components "above" the page (like Material) and the "push away" approach

5. I would like to see a simplified top-bar ;)

6. CSS animations? Mhh...maybe as a separate component/plugin

7. full screen modals (like this http://git.aaronlumsden.com/tidytime.js/#demo click the demo button)

8, +1 for datepicker

9. dropdowns could use some love.... the component looks not well refined

10. http://lab.hakim.se/ladda/ ?

11. (OT) a repository for snippets like http://bootsnipp.com/?page=1 ?

**Comment by gakimball** [comment]
@erredeco Thanks for your feedback! :)

As far as typography goes, we're looking at employing a modular scale for typography. We also might look at a responsive scale system, where the scale used to size type is compressed on smaller screens.

Button classes aren't likely to go; buttons are a very flexible component so having all of the sizing and styling classes keeps them easy to use. This framework still has a CSS download so we need those helper classes still.

We had a good chat about dropdowns last week and we have some ideas to make them more flexible. Same with top bar.

**Comment by erredeco** [comment]
@gakimball my question was: If I only use standard and  .alert button, why should I have in the css also .success, .secondary, .info ? isn't it a waste? The alternative could be that _buttons.scss does not include the button mixin, so I can build only the buttons I need :)  Or that there is a boolean flag like $include-full-button-classes or whatever

**Comment by erredeco** [comment]
It comes in my mind another great addition: a "tile" component a là Windows8, maybe with animations like flip, fade... what do you think?

**Comment by rafibomb** [comment]
@Anhellius - Any luck on a mega menu, really curious how you would handle the mobile part.

**Comment by rafibomb** [comment]
@ksherman Yes, equalizer can do this with an equalize_on_stack option.

**Comment by 3lfist** [comment]
Would love to have a `bottom-bar` component to go along with the `top-bar`. This would be great for web apps access on smart phones and tablets. :+1:

**Comment by Lynda333** [comment]
Recently I signed up for a new service. The new service took me through a tour of the dashboard, very similar to how Joyride works, but there was one feature of their tour I really liked that I would like to see built into Foundation's Joyride. That feature was a highlighting feature. In other words, when a tip is pointing to a certain div everything but the containing div had an overlay over it. It really helped isolate and show off the div that the modal was highlighting. Hope that makes sense.

**Comment by erredeco** [comment]
@Lynda333 

something like: http://jquerytools.github.io/demos/toolbox/expose/index.html?

**Comment by Lynda333** [comment]
@erredeco - Yup, that is what I am speaking of, just tied into Joyride.

**Comment by b3nl** [comment]
@erredeco @Lynda333 +1

**Comment by Lynda333** [comment]
One other thought in reference to the TopBar and Dropdowns. I know you are looking at rebuilding the topbar, when you do please have an option that if a person mouses out of a dropdown we can set how long before it closes. CSS Tricks had a good article [talking](http://css-tricks.com/dropdown-menus-with-more-forgiving-mouse-movement-paths/) about this.

**Comment by benedictleejh** [comment]
I've been mostly happy with Foundation, and I'm by and large happy with what's available in terms of components. I do, however, have a couple of things I'd like to see:

1. A generic sticky functionality that can be attached to any component (within reason; a sticky on a modal doesn't quite make much sense to me, for example). Right now, as far as I can tell, Magellan only works with a sub-nav, but there might be times I'd want to have a sticky button, or sidebar, or something. I've yet to play around with Magellan much, but it was one of the first thoughts I had when I saw the Magellan docs.

2. A more flexible off-canvas. I use the off-canvas quite a bit in my app (not as a menu), and I've always wanted more flexibility, For example, being able to have it pop out of either the top, bottom, left, or right; individually specified sizes, background colours, etc. for each off-canvas; multiple off-canvas on each side (might make for a cool sidebar of buttons that each open a different menu or something). I understand some of it is doable with Foundation for Apps, but I'm not keen on bringing Angular in as a dependency (I have enough dependencies as it is).

Theming is an interesting idea. I'm generally happy with the Foundation defaults, though I'd be interested in playing around with something like a Material Design theme if Foundation had one. Related to this, though, it'd be nice if the rails gem generated multiple SCSS files for the Foundation variables; right now, it's a huge file, which makes it a little intimidating to go through.

**Comment by rainbowshrimp** [comment]
I think a date picker and autocomplete would make good additions to the framework.

**Comment by darkwoods-** [comment]
Top-bar replaced with something much, much better should be a definite!

as for the block grid merging it into the grid means what exactly?
like it wont be in ul list anymore?

**Comment by SalahAdDin** [comment]
I use the top bar menu, i think that this is very useful, many web page use top bar menues, so, i think that won't be deprecated. :+1:

**Comment by tjhiggins** [comment]
Having an autocomplete would be the final piece for us to stop using jQuery UI.

**Comment by Renkas** [comment]
Would love to have retina background images support. Something like Bourbon has: http://bourbon.io/docs/#retina-image

**Comment by zurbrandon** [comment]
Thanks for the feedback, all! We've looked it over and are really rocking away at getting Foundation 6 out very soon!
